ui/ducks/domains.js

```diff
@@ -188,18 +202,56 @@ export function lookupEnsName(domainName) {
         error = err;
       }
       const chainId = getCurrentChainId(state);
-      const network = CHAIN_ID_TO_NETWORK_ID_MAP[chainId];
-
-      await dispatch(
-        domainLookup({
-          address,
-          error,
-          chainId,
-          network,
-          domainType: ENS,
-          domainName: trimmedDomainName,
-        }),
-      );
+      let network = CHAIN_ID_TO_NETWORK_ID_MAP[chainId];
+      const nameLookupSnaps = getNameLookupSnapsIds(state);
+      if (nameLookupSnaps.length > 0) {
+        const results = await Promise.all(
+          nameLookupSnaps.map((snapId) => {
+            return new Promise((resolve) => {
+              handleSnapRequest({
+                snapId,
+                origin: '',
+                handler: 'onNameLookup',
+                request: {
+                  jsonrpc: '2.0',
+                  method: ' ',
+                  params: {
+                    domain: domainName,
+                    chainId: `eip155:${parseInt(chainId, 10)}`,
+                  },
+                },
+              }).then((res) => {
+                resolve(res);
+              });
+            });
+          }),
+        );
+        const successfulResolutions = results.filter(
+          (result) => result !== null,
+        );
+        network = parseInt(chainId, 10);
+        await dispatch(
+          domainLookup({
+            address: successfulResolutions[0].resolvedAddress,
+            error,
+            chainId,
+            network,
+            domainType: 'Other',
+            domainName: trimmedDomainName,
+          }),
+        );
```

Comment:
Could this be extracted in it's own function/file ?

Reply:
Yes, that's the aim of `useDomainResolution`, haven't finished it up since this PR was blocked pending release.